### PR TITLE
many: switch to use http numeric statuses as agreed

### PR DESCRIPTION
**Pull request by pedronis**
Also change the code that was interpreting a 401 as "please buy" which is the unlikely case.

Also have a static check to make sure we continue with this style.

### CODE REVIEW

File: store/store_test.go [hunk 1]
```diff
@@ -698,15 +700,15 @@ func (t *remoteRepoTestSuite) TestActualDownloadNonPurchased401(c *C) {
 	var buf bytes.Buffer
 	err := download(context.TODO(), "foo", "sha3", mockServer.URL, nil, theStore, nopeSeeker{&buf}, -1, nil)
 	c.Assert(err, NotNil)
-	c.Check(err.Error(), Equals, "Please buy foo before installing it.")
+	c.Check(err.Error(), Equals, "please buy foo before installing it.")
```
Comment (chipaca) [inline store/store_test.go]: nice drive-by there

File: run-checks [hunk 2]
```diff
@@ -110,6 +110,21 @@ if [ "$STATIC" = 1 ]; then
     echo Running vet
     go list ./... | grep -v '/vendor/' | xargs go vet
 
+    echo 'Check for usages of http.Status*'
+    got=""
+    for dir in $(go list -f '{{.Dir}}' ./... | grep -v '/vendor/' ); do
+        s="$(grep -nE "http\.Status"  "$dir"/*.go |grep -vE "http\.StatusText\(" || true)"
```
Comment (chipaca) [inline run-checks]: you could also write that `grep -nP 'http\.Status(?!Text)'`, fwiw.

File: store/store_test.go [hunk 3]
```diff
@@ -685,11 +685,13 @@ func (nopeSeeker) Seek(int64, int) (int64, error) {
...
 }
 
-func (t *remoteRepoTestSuite) TestActualDownloadNonPurchased401(c *C) {
+func (t *remoteRepoTestSuite) TestActualDownloadNonPurchased402(c *C) {
 	n := 0
 	mockServer := httptest.NewServer(http.HandlerFunc(func(w http.ResponseWriter, r *http.Request) {
 		n++
-		w.WriteHeader(http.StatusUnauthorized)
+		// XXX: the server doesn't behave correctly ATM
+		// but 401 for non purchases is the unlikely case so far
```
Comment (chipaca) [inline store/store_test.go]: is this still true? I thought work was being done on this front

Comment (pedronis) [inline store/store_test.go]: yes, is still true to some extent
